### PR TITLE
upgrades && typescript migration

### PR DESCRIPTION
**Pull request by ndelangen**
builds upon https://github.com/storybookjs/design-system/pull/164

### CODE REVIEW

File: src/components/AvatarList.stories.js [hunk 1]
```diff
@@ -35,19 +35,10 @@ export default {
...
-
-export const smallSize = () => <AvatarList users={users} userCount={100} size="small" />;
-
-smallSize.story = {
-  name: 'small size',
-};
-
-export const loading = () => <AvatarList isLoading />;
-export const empty = () => <AvatarList users={[]} />;
+export const Basic = argsStory({ users });
```
Comment (kylesuss) [inline src/components/AvatarList.stories.js:38]: @ndelangen @shilman is this the same thing being discussed here: https://github.com/storybookjs/design-system/pull/164#discussion_r421406969? Seems like there is a bit of overlap in between these 2 PRs so not sure if this new PR from @ndelangen cancels the other one out or what.

Comment (ndelangen) [reply src/components/AvatarList.stories.js:38]: I've used @shilman's work as a base-branch.

That PR should be discussed and merged first.

Comment (ndelangen) [reply src/components/AvatarList.stories.js]: Ready

File: .storybook/main.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
   stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.js'],
   addons: [
+    '@storybook/preset-typescript',
```
Comment (shilman) [inline .storybook/main.js:4]: Why do we need this? Zero config?

File: package.json [hunk 3]
```diff
@@ -32,74 +32,68 @@
...
+    "@babel/preset-react": "^7.10.1",
+    "@babel/preset-typescript": "^7.10.1",
     "@storybook/addon-a11y": "6.0.0-beta.20",
     "@storybook/addon-actions": "6.0.0-beta.20",
     "@storybook/addon-docs": "6.0.0-beta.20",
     "@storybook/addon-essentials": "6.0.0-beta.20",
     "@storybook/addon-storysource": "6.0.0-beta.20",
+    "@storybook/eslint-config-storybook": "^2.4.0",
+    "@storybook/linter-config": "^2.5.0",
+    "@storybook/preset-typescript": "^3.0.0",
```
Comment (shilman) [inline package.json:70]: Nope

File: package.json [hunk 4]
```diff
@@ -32,74 +32,68 @@
...
-    "prettier-eslint": "^9.0.0",
-    "react": "^16.8.3",
-    "react-dom": "^16.8.3",
-    "sort-package-json": "^1.21.0",
-    "storybook-chromatic": "^3.0.3"
+    "babel-plugin-styled-components": "^1.10.7",
+    "chromatic": "^5.0.0-alpha.4",
+    "cross-env": "^7.0.2",
+    "eslint": "^7.1.0",
+    "fork-ts-checker-webpack-plugin": "^4.1.6",
```
Comment (shilman) [inline package.json:81]: Nope?

File: .storybook/main.js (hunk 2 above)
Comment (ndelangen) [reply .storybook/main.js:4]: I can delete this? nice!

File: package.json (hunk 4 above)
Comment (ndelangen) [reply package.json:81]: cool cool

**Comment by ndelangen** [comment]
:rocket: PR was released in v5.1.0 :rocket:

**Pull request merged**